export/mlab_export.py

```diff
@@ -0,0 +1,613 @@
+    'Timestamp to end export, in seconds since the epoch. Only use this option '
+    'for debugging. Normally, ts_end is calculated automatically from: '
+    'ts_start + length.', lower_bound=0)
+flags.DEFINE_integer(
+    'ts_offset', 600, 'Amount of time (seconds) that must have passed after '
+    'ts_end to ensure that values cached by collectd have been flushed to disk '
+    'before attempting an export.', lower_bound=0)
+flags.DEFINE_multistring(
+    'ignored_experiments', [], 'List of experiment names to ignore. Experiment '
+    'must be in "slice.site" form not "site_slice".')
+flags.DEFINE_bool(
+    'pretty_json', None, 'Add extra indenting to json output (for debugging).')
+flags.DEFINE_string(
+    'output_dir', EXPORT_DIR, 'Root directory of json output files.')
+flags.DEFINE_string(
+    'output', None, 'Name of json output file. Set automatically if not given.')
+flags.DEFINE_string(
+    'export_metrics', METRIC_MAP_CONF, 'File name with metric map. The metric '
+    'map defines canonical metric names for raw, metric names taken from '
+    'collectd RRD files.')
+flags.DEFINE_bool(
+    'verbose', False, 'Increase verbosity level.')
+flags.DEFINE_bool(
+    'show_nagios', False, 'Shows collectd-nagios commands to monitor metrics.')
+flags.DEFINE_bool(
+    'show_rrdfile', False, 'Shows the RRD files opened during export.')
+flags.DEFINE_bool(
+    'show_metric', False, 'Shows the canonical metric names during export.')
+flags.DEFINE_bool(
+    'show_rawmetric', False, 'Shows the raw metric name before translating to '
+    'the canonical name. This can be helpful to know before adding new metrics '
+    'to the export metrics configuration file.')
+flags.DEFINE_bool(
+    'show_skipped', False, 'Shows the raw metric names that are not exported. '
+    'This option may be helpful when adding new metrics to the export_metrics.')
+flags.DEFINE_bool(
+    'update', True, 'Update timestamps on successful export. Update is always '
+    'disabled when any --show_* option is enabled.')
+flags.DEFINE_bool(
+    'compress', False,
+    'Compresses output and adds .gz extension to output filename.')
+
+
+class Error(Exception):
+  """Base error type for this file."""
+
+
+class TimeOptionError(Error):
+  """An error related to export times or ranges."""
+
+
+class LockFileError(Error):
+  """An error related to the single process lock file."""
+
+
+def init_global():
+  global HOSTNAME
+  # NOTE: This should be the hostname of root context, not slice context.
+  HOSTNAME = socket.gethostname()
+  logging.basicConfig(format='%(message)s', level=logging.INFO)
+
+
+class LockFile(object):
+  """Provides a file-based lock."""
+
+  def __init__(self, filename):
+    self._filename = filename
+    self._handle = None
+
+  def __enter__(self):
+    """Acquires file lock on filename. Raises IOError on failure."""
+    self._handle = open(self._filename, 'w')
+    try:
+      fcntl.flock(self._handle, fcntl.LOCK_EX|fcntl.LOCK_NB)
```

Comment:
I think there's supposed to be a single space on either side of the pipe since it's a binary operator, though style guide is not 100% clear in this case:
https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Whitespace#Whitespace